src/tests/async/shared-generic.cs

```diff
@@ -6,16 +6,51 @@
 using Xunit;
 public class Async2SharedGeneric
 {
+    public struct S0
+    {
+        public object _o;
+        public S0(object o) => _o = o;
+    }
+
+    public struct S1<T>
+    {
+        public T t;
+    }
+
     [Fact]
     public static void TestEntryPoint()
     {
+        // simple cases
         Async1EntryPoint<int>(typeof(int), 42).Wait();
         Async1EntryPoint<string>(typeof(string), "abc").Wait();
         Async1EntryPoint<object>(typeof(object), "def").Wait();
 
+        // struct with an obj and its nullable
+        Async1EntryPoint<S0>(typeof(S0), new S0(42)).Wait();
+        // TODO: uncomment to repro https://github.com/dotnet/runtimelab/issues/3043
+        // Async1EntryPoint<S0?>(typeof(S0?), new S0(42)).Wait();
```

Comment:
Logged an issue (https://github.com/dotnet/runtimelab/issues/3043) to track this failure. 

I do not think it is related to the returning a struct. In this case we start with non-null value, but lose the value right after resuming and then we return null. 
It may not be related to Nullable/boxing at all, or related only indirectly.